### PR TITLE
 Added GitHub Flavoured markdown cheatsheet

### LINKED ISSUE #1906: Convert Markdown CheatSheet to new format, add support for GitHub Flavored MD
We have an existing "Markdown Cheat Sheet" that doesn't actually trigger for the query "Markdown Cheat Sheet" because it is implemented as a Perl Goodie that triggers on specific markdown elements:

https://duckduckgo.com/?q=markdown+header&t=ffab&ia=answer

We should probably have a JSON Cheat Sheet for Markdown. As mentioned in https://github.com/duckduckgo/zeroclickinfo-fathead/issues/167 we should also support "GitHub Flavoured Markdown". It could be a separate Cheat sheet or combined into the new Markdown Cheat Sheet

### PR DESCRIPTION
**Pull request by gautamkrishnar**
Fixes #1906 

-----
IA Page: https://duck.co/ia/view/github_flavoured_markdown_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata post coming soon

File: share/goodie/cheat_sheets/json/github_flavoured_markdown.json [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    ],
+    "Syntax highlighting": [
+      {
+        "key": "```ruby ... code block ```",
+        "val": "Code blocks can be taken a step further by adding syntax highlighting. In your fenced block, add an optional language identifier"
+      }
+    ],
+    "Tables": [
+      {
+        "key": "First Header  | Second Header\n------------- | -------------\nContent Cell  | Content Cell\nContent Cell  | Content Cell",
```
Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/github_flavoured_markdown.json]: @moollaza It should be great if we can parse \n as a newline

**Comment by gautamkrishnar** [comment]
cc: @moollaza

File: share/goodie/cheat_sheets/json/github_flavoured_markdown.json [hunk 2]
```diff
@@ -0,0 +1,68 @@
+{
+  "id": "github_flavoured_markdown_cheat_sheet",
+  "name": "Github Flavoured Markdown",
+  "description": "GitHub uses GFM, across the site in issues, comments, and pull requests. It differs from standard Markdown in a few significant ways, and adds some additional functionality.",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/github_flavoured_markdown.json]: @gautamkrishnar  I think this can be concise.

Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/github_flavoured_markdown.json]: nirtheeettu podey :smile:

Comment (jophab) [inline share/goodie/cheat_sheets/json/github_flavoured_markdown.json]: @gautamkrishnar Its too long man...
Maryadakku maattikkoo... :smile:

Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/github_flavoured_markdown.json]: Oooh sheri cherukkaaa

Comment (moollaza) [inline share/goodie/cheat_sheets/json/github_flavoured_markdown.json]: We should always aim to describe the cheat sheet and it's content:

More along the lines of: `"Overview of syntax and common elements for GitHub Flavoured Markdown, a special form of Markdown used across GitHub"

File: share/goodie/cheat_sheets/json/github_flavoured_markdown.json [hunk 3]
```diff
@@ -0,0 +1,68 @@
+{
+  "id": "github_flavoured_markdown_cheat_sheet",
+  "name": "Github Flavoured Markdown",
+  "description": "GitHub uses GFM which is different from standard Markdown in a few significant ways, and adds some additional functionality.",
+  "metadata": {
+    "sourceName": "GitHub",
+    "sourceUrl": "https://help.github.com/articles/github-flavored-markdown/"
+  },
+  "aliases": [
+    "ghfm",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/github_flavoured_markdown.json]: would add `gfm` as that's the common short form

File: share/goodie/cheat_sheets/json/github_flavoured_markdown.json [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    "ghfmd"
+  ],
+  "template_type": "terminal",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/github_flavoured_markdown.json]: `code` would be better here

**Comment by gautamkrishnar** [comment]
@moollaza :+1:

File: share/goodie/cheat_sheets/json/github_flavoured_markdown.json [hunk 5]
```diff
@@ -0,0 +1,68 @@
+{
+  "id": "github_flavoured_markdown_cheat_sheet",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/github_flavoured_markdown.json]: @gautamkrishnar the filename is incorrect so this IA isn't triggering.

You need to replace the `_` with `-`

File: share/goodie/cheat_sheets/json/github_flavoured_markdown.json [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  "aliases": [
+    "gfm",
+    "ghfmd"
+  ],
+  "template_type": "code",
+  "section_order": [
+    "Multiple underscores in words",
+    "URL autolinking",
+    "Strikethrough",
+    "Fenced code blocks",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/github_flavoured_markdown.json]: Let's Title Case the section names

**Comment by gautamkrishnar** [comment]
@moollaza  done :+1:

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Github Flavoured Markdown Cheat Sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/github_flavoured_markdown_cheat_sheet_mobile.png)

**Comment by gautamkrishnar** [comment]
@moollaza still having the same problem:
screenshot 139

**Comment by gautamkrishnar** [comment]
@moollaza hope you forgot this one.. :smile:

File: share/goodie/cheat_sheets/json/github-flavoured-markdown.json [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+  "description": "Overview of syntax and common elements for GitHub Flavoured Markdown, a special form of Markdown used across GitHub",
+  "metadata": {
+    "sourceName": "GitHub",
+    "sourceUrl": "https://help.github.com/articles/github-flavored-markdown/"
+  },
+  "aliases": [
+    "gfm",
+    "ghfmd"
+  ],
+  "template_type": "code",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/github-flavoured-markdown.json]: Looks like `terminal` would be better here

File: share/goodie/cheat_sheets/json/github-flavoured-markdown.json [hunk 8]
```diff
@@ -0,0 +1,68 @@
+{
+  "id": "github_flavoured_markdown_cheat_sheet",
+  "name": "Github Flavoured Markdown",
+  "description": "Overview of syntax and common elements for GitHub Flavoured Markdown, a special form of Markdown used across GitHub",
+  "metadata": {
+    "sourceName": "GitHub",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/github-flavoured-markdown.json]: indentation is off too, 4-space soft tabs :+1:

**Comment by moollaza** [comment]
@gautamkrishnar Thanks! This LGTM :+1:

I'll fix the indentation and template.

**Pull request merged**